packages/app/components/Nav/NotificationSettingsModal.tsx

```diff
@@ -125,3 +121,90 @@ export function NotificationSettingsModal({
     </Modal>
   );
 }
+
+function useThisDeviceEndpoint(): null | string {
+  const [endpoint, setEndpoint] = useState(null);
+  useEffect(() => {
+    (async () => setEndpoint(await getEndpoint()))();
+  });
+  return endpoint;
+}
+
+function renderDeviceInfo(
+  device: DesktopNotifPartial,
+  isThisDevice: boolean
+): string {
+  if (device.name) {
+    return isThisDevice ? `${device.name} (This Device)` : device.name;
+  } else {
+    return isThisDevice ? "This Device" : "Other Device";
+  }
```

Comment:
Just curious when would we reach this `else` state? Is it for the legacy desktop notifications?

Reply:
yep for legacy!